ffi/diplomat/src/decimal.rs

```diff
@@ -76,22 +76,16 @@ pub mod ffi {
             data_struct: &mut ICU4XDataStruct,
             options: ICU4XFixedDecimalFormatOptions,
         ) -> DiplomatResult<Box<ICU4XFixedDecimalFormat>, ()> {
-            let data = if let Some(data) = data_struct.0.take() {
-                data
-            } else {
-                return Err(()).into();
-            };
-
-            let data = if let Ok(data) = data.downcast::<DecimalSymbolsV1Marker>() {
-                data
-            } else {
-                return Err(()).into();
-            };
-            let provider = StructProvider {
+            use icu_provider::prelude::AsDowncastingAnyProvider;
+            let provider = AnyPayloadProvider {
                 key: SYMBOLS_V1,
-                data,
+                data: data_struct.0.clone(),
```

Comment:
question: any reason to not use `.take()`? I was trying to avoid the extra clone here.

Alternatively, if we're using `.clone()`, we should make `data_struct` an immutable borrow

Reply:
Okay, I see.  Hmm.  I think we were already performing a clone, since `impl DataProvider for StructProvider` also clones the data (both before and after this change).  Note that the clone here is cheap, because it's incrementing the `Rc`.  We don't actually clone any real data until we `.downcast()`, which happens in `impl DataProvider for AnyPayloadProvider`.

Reply:
I see two choices:

1. Add a version of AnyPayloadProvider that has an internal `Cell` that it clears out when data is fetched from it, in order to avoid the clone, or
2. Stick with the design of data providers handing out clones of DataPayloads (which are generally cheap, except for cases where the data struct is manually generated, like it is here), and then change this function to take the ICU4XStructProvider by immutable reference

Reply:
I think (2) is fine. Make sure to update the docs of this function and ICU4XDataStruct. We can also get rid of the `Option` in ICU4XDataStruct.

Reply:
I did (2), and I had already previously remove the `Option`.